database/pgx/pgx.go

```diff
@@ -394,37 +401,68 @@ func (p *Postgres) Version() (version int, dirty bool, err error) {
 }
 
 func (p *Postgres) Drop() (err error) {
-	// select all tables in current schema
-	query := `SELECT table_name FROM information_schema.tables WHERE table_schema=(SELECT current_schema()) AND table_type='BASE TABLE'`
-	tables, err := p.conn.QueryContext(context.Background(), query)
+	// drop all tables in current schema
+	err = p.dropElement(tableSQL)
+	if err != nil {
+		return err
+	}
+
+	// drop all custom types in current schema
+	err = p.dropElement(typeSQL)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Postgres) dropElement(element sqlElement) (err error) {
+	// select all of "element" in current schema
+	var query string
+	switch element {
+	case tableSQL:
+		query = `SELECT table_name FROM information_schema.tables WHERE table_schema=(SELECT current_schema()) AND table_type='BASE TABLE'`
+	case typeSQL:
```

Comment:
Add tests for this. e.g. that this will only drop unused types in the current schema

Reply:
This function does not check for unused. `Drop()` however does what you expect it would. First it drops all the tables. Than it drops _all_ the custom types. Because the tables are dropped first all the custom types are unused.
The expected result is now tested.